tests/foreman/api/test_errata.py

```diff
@@ -823,3 +832,130 @@ def test_positive_incremental_update_apply_packages_to_envs_cvs(self):
+        cls.org = entities.Organization().create()
+        cls.lce = entities.LifecycleEnvironment(organization=cls.org).create()
+        cls.repos_collection = RepositoryCollection(
+            distro=DISTRO_RHEL8,
+            repositories=[
+                YumRepository(url=CUSTOM_SWID_TAG_REPO),
+            ]
+        )
+        cls.repos_collection.setup_content(
+            cls.org.id,
+            cls.lce.id,
+            upload_manifest=True
+        )
+
+    def _run_remote_command_on_content_host(self, command, vm, return_result=False):
+        result = vm.run(command)
+        self.assertEquals(result.return_code, 0)
+        if return_result:
+            return result.stdout
+
+    def _set_prerequisites_for_swid_repos(self, vm):
+        self._run_remote_command_on_content_host(
+            "wget --no-check-certificate {}".format(settings.swid_tools_repo),
+            vm)
+        self._run_remote_command_on_content_host("mv *swid*.repo /etc/yum.repos.d", vm)
+        self._run_remote_command_on_content_host("yum install -y swid-tools", vm)
+        self._run_remote_command_on_content_host("dnf install -y dnf-plugin-swidtags", vm)
+
+    def _validate_swid_tags_installed(self, vm, module_name):
+        result = self._run_remote_command_on_content_host(
+            "swidq -i -n {} | grep 'Name'".format(module_name),
+            vm,
+            return_result=True)
+        self.assertIn(module_name, result)
+
+    @tier3
+    @upgrade
+    def test_errata_installation_with_swidtags(self):
+        """Verify errata installation with swid_tags and swid tags get updated after
+        module stream update.
+
+        :id: 43a59b9a-eb9b-4174-8b8e-73d923b1e51e
+
+        :steps:
+
+            1. create product and repository having swid tags
+            2. create content view and published it with repository
+            3. create activation key and register content host
+            4. create rhel8, swid repos on content host
+            5. install swid-tools, dnf-plugin-swidtags packages on content host
+            6. install older module stream and generate errata, swid tag
+            7. assert errata count, swid tags are generated
+            8. install errata vis updating module stream
+            9. assert errata count and swid tag after module update
+
+        :expectedresults: swid tags should get updated after errata installation via
+            module stream update
+
+        :CaseAutomation: Automated
+
+        :CaseImportance: Critical
+
+        :CaseLevel: System
+        """
+        with VirtualMachine(distro=self.repos_collection.distro) as vm:
+            module_name = 'kangaroo'
+            version = '20180704111719'
+            # setup rhel8 and sat_tools_repos
+            for repo_name in ('baseos', 'appstream'):
+                vm.create_custom_repos(**{
+                    repo_name: settings.rhel8_os[repo_name]
+                })
```

Comment:
You don't have to create dict only to unpack it. You can just pass named argument to function:
```
vm.create_custom_repos(repo_name=settings.rhel8_os[repo_name])
```

Reply:
Above code is not working for me, this leads an error test. We are replacing the same file again as the repo_name variable is passed as value repo_name instead passing repo_name's value. 

```
2019-05-27 15:26:07 - robottelo.ssh - INFO - >>> echo "[repo_name]
name=repo_name
baseurl=http://{URL}/BaseOS/x86_64/os/
enabled=1
gpgcheck=0" > /etc/yum.repos.d/repo_name.repo
2019-05-27 15:26:09 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fab42a02e10
2019-05-27 15:26:12 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fab429f02e8
2019-05-27 15:26:12 - robottelo.ssh - INFO - Connected to [10.*.*.*]
2019-05-27 15:26:12 - robottelo.ssh - INFO - >>> echo "[repo_name]
name=repo_name
baseurl=http://{URL}/AppStream/x86_64/os/
enabled=1
gpgcheck=0" > /etc/yum.repos.d/repo_name.repo
```

Reply:
You are right, the way I proposed one repo will overwrite another. If you want to pass variable content as argument name, you have to use unpacking.

But it seems that you don't need do that in the loop. Can you try if this will work?
```
vm.create_custom_repos(**{
   repo_name: settings.rhel8_os[repo_name] for repo_name in ('baseos', 'appstream')
})
```

Reply:
@mirzal worked!